pkg/api/customization/project/project_actions.go

```diff
@@ -0,0 +1,147 @@
+package project
+
+import (
+	"fmt"
+
+	"net/http"
+
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/parse"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	"github.com/rancher/types/client/management/v3"
+)
+
+func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+	resource.AddAction(apiContext, "setpodsecuritypolicytemplate")
+}
+
+type Handler struct {
+	ProjectClient v3.ProjectInterface
+}
+
+func (h *Handler) Actions(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	switch actionName {
+	case "setpodsecuritypolicytemplate":
+		return h.setPodSecurityPolicyTemplate(actionName, action, apiContext)
+	}
+
+	return errors.Errorf("bad action %v", actionName)
+}
+
+func (h *Handler) setPodSecurityPolicyTemplate(actionName string, action *types.Action, request *types.APIContext) error {
+	actionInput, err := parse.ReadBody(request.Request)
```

Comment:
can you follow this paradigm for getting and parsing the action body?
https://github.com/rancher/rancher/blob/master/pkg/auth/providers/activedirectory/actions.go#L41-L50

It will give you free validation of the body (like ensuring required fields are present and the types of the fields are correct). This action is simple, so it is not the most important thing for this particualr instance, but I want to make it the standard way of parsing action bodies

Reply:
Note that in that example, I unmarshal to a map but then also convert to the struct type. You do not need to do the second step if you dont want to. You can work with the map if you want. My action had a more complex set of inputs, so working with the actual type was easier